tests/test_utils.py

```diff
@@ -488,3 +500,60 @@ def test_get_error_message(error, expected_result):
         exc_value.detail = error
         raise Exception
     assert get_error_message(exc_value) == expected_result(exc_value)
+
+
+def test_installed_modules():
```

Comment:
It looks to me as though this test is simply generating the installed module information in the same or a similar way that the actual code does it in `utils.py`. I would prefer that we somehow compare against a manually generated output, though, since this test will fail to detect a case where something changes in the `importlib` or `pkg_resources` modules that causes the test and the actual code to break in the same way, and it is also more difficult by reading this test case to determine what the intended output is. 

Perhaps we can somehow mock out the installed modules so that we can manually generate the expected output, but admittedly doing this may be somewhat complicated. I also don't mind if we delete this test case, or replace it with a test case that simply calls the function and ensures we don't get an error, since I am unsure that it adds much value since we are just repeating the code from the function we are testing.

Reply:
So this test is a bit unconventional. On py versions where only one of `importlib.metadata`, `pkg_resources` exists, it's exactly as you write -- it isn't testing much. It only makes sense on py versions that have both `importlib.metadata` and `pkg_resources`, since in that case it compares whether our ways of getting and normalizing the package name yield the same result for packages from both `importlib.metadata` and `pkg_resources`.